### DISCUSSION TITLE
Mandatory fields for onest:financial-support domain

### DISCUSSION THREAD

**Discussion by nirmalnr** [ONEST-Network/community#29] [General]
Hi all,

Sharing the proposed mandatory fields for financial support. Please propose if any changes are required.

API | Field
-- | --
search | context.domain
search | context.action
search | context.version
search | context.bap_id
search | context.bap_uri
search | context.transaction_id
search | context.message_id
search | context.timestamp
on_search | context.domain
on_search | context.action
on_search | context.timestamp
on_search | context.version
on_search | context.bap_id
on_search | context.bap_uri
on_search | context.bpp_id
on_search | context.bpp_uri
on_search | context.transaction_id
on_search | context.message_id
on_search | message.catalog.descriptor.name
on_search | message.catalog.providers[].id
on_search | message.catalog.providers[].descriptor.name
on_search | message.catalog.providers[].items[].id
on_search | message.catalog.providers[].items[].descriptor.name
on_search | message.catalog.providers[].items[].descriptor.long_desc
on_search | message.catalog.providers[].items[].price.currency
on_search | message.catalog.providers[].items[].price.value
select | context.domain
select | context.action
select | context.timestamp
select | context.version
select | context.bap_id
select | context.bap_uri
select | context.bpp_id
select | context.bpp_uri
select | context.transaction_id
select | context.message_id
select | message.order.items[].id
select | message.order.provider.id
on_select | context.domain
on_select | context.action
on_select | context.timestamp
on_select | context.ttl
on_select | context.version
on_select | context.bap_id
on_select | context.bap_uri
on_select | context.bpp_id
on_select | context.bpp_uri
on_select | context.transaction_id
on_select | context.message_id
on_select | message.order.provider.id
on_select | message.order.provider.descriptor.name
on_select | message.order.items[].id
on_select | message.order.items[].descriptor.name
on_select | message.order.items[].descriptor.long_desc
init | context.domain
init | context.action
init | context.timestamp
init | context.version
init | context.bap_id
init | context.bap_uri
init | context.bpp_id
init | context.bpp_uri
init | context.transaction_id
init | context.message_id
init | message.order.items[].id
init | message.order.provider.id
init | message.order.fulfillments[].customer.person.name
init | message.order.fulfillments[].customer.person.age
init | message.order.fulfillments[].customer.person.gender
init | message.order.fulfillments[].customer.contact.phone
init | message.order.fulfillments[].customer.contact.email
on_init | context.domain
on_init | context.action
on_init | context.timestamp
on_init | context.version
on_init | context.bap_id
on_init | context.bap_uri
on_init | context.bpp_id
on_init | context.bpp_uri
on_init | context.transaction_id
on_init | context.message_id
on_init | message.order.provider.id
on_init | message.order.provider.descriptor.name
on_init | message.order.items[].id
on_init | message.order.items[].descriptor.name
on_init | message.order.items[].descriptor.long_desc
on_init | message.order.items[].fulfillment_ids[]
on_init | message.order.fulfillments[].state.descriptor.code
on_init | message.order.fulfillments[].state.descriptor.name
on_init | message.order.fulfillments[].state.updated_at
on_init | message.order.fulfillments[].id
on_init | message.order.fulfillments[].customer.person.name
on_init | message.order.fulfillments[].customer.person.age
on_init | message.order.fulfillments[].customer.person.gender
on_init | message.order.fulfillments[].customer.contact.phone
on_init | message.order.fulfillments[].customer.contact.email
on_init | context.timestamp
on_init | context.location.city.name
on_init | context.location.city.code
on_init | context.location.country.name
on_init | context.location.country.code
confirm | context.domain
confirm | context.action
confirm | context.timestamp
confirm | context.version
confirm | context.bap_id
confirm | context.bap_uri
confirm | context.bpp_id
confirm | context.bpp_uri
confirm | context.transaction_id
confirm | context.message_id
confirm | message.order.items[].id
confirm | message.order.provider.id
confirm | message.order.fulfillments[].customer.person.name
confirm | message.order.fulfillments[].customer.person.age
confirm | message.order.fulfillments[].customer.person.gender
confirm | message.order.fulfillments[].customer.contact.phone
confirm | message.order.fulfillments[].customer.contact.email
on_confirm | context.domain
on_confirm | context.action
on_confirm | context.timestamp
on_confirm | context.version
on_confirm | context.bap_id
on_confirm | context.bap_uri
on_confirm | context.bpp_id
on_confirm | context.bpp_uri
on_confirm | context.transaction_id
on_confirm | context.message_id
on_confirm | message.order.id
on_confirm | message.order.provider.id
on_confirm | message.order.provider.descriptor.name
on_confirm | message.order.items[].id
on_confirm | message.order.items[].descriptor.name
on_confirm | message.order.items[].descriptor.long_desc
on_confirm | message.order.fulfillments[].state.descriptor.code
on_confirm | message.order.fulfillments[].state.descriptor.name
on_confirm | message.order.fulfillments[].state.updated_at
on_confirm | message.order.fulfillments[].id
on_confirm | message.order.fulfillments[].customer.person.name
on_confirm | message.order.fulfillments[].customer.person.age
on_confirm | message.order.fulfillments[].customer.person.gender
on_confirm | message.order.fulfillments[].customer.contact.phone
on_confirm | message.order.fulfillments[].customer.contact.email
status | context.domain
status | context.action
status | context.timestamp
status | context.version
status | context.bap_id
status | context.bap_uri
status | context.bpp_id
status | context.bpp_uri
status | context.transaction_id
status | context.message_id
status | message.order_id
on_status | message.order.id
on_status | message.order.provider.id
on_status | message.order.provider.descriptor.name
on_status | message.order.items[].id
on_status | message.order.items[].descriptor.name
on_status | message.order.items[].descriptor.long_desc
on_status | message.order.fulfillments[].state.descriptor.code
on_status | message.order.fulfillments[].state.descriptor.name
on_status | message.order.fulfillments[].state.updated_at
on_status | message.order.fulfillments[].id
on_status | message.order.fulfillments[].customer.person.name
on_status | message.order.fulfillments[].customer.person.age
on_status | message.order.fulfillments[].customer.person.gender
on_status | message.order.fulfillments[].customer.contact.phone
on_status | message.order.fulfillments[].customer.contact.email

**Comment by nikhil-scholarlify** [comment]
@nirmalnr @shiva-rakshith  query:  are these required to be in integer, float or string format?
```
message.catalog.providers[].items[].price.value
OR
message.catalog.providers[].items[].price.minimum_value
message.catalog.providers[].items[].price.maximum_value
```
as given in https://github.com/ONEST-Network/ONEST-Specification/blob/main/docs/implementation_guides/financial-support/mandatory_fields.md 

Suggestion: add a column for indicating expected / allowed formats, or add a note at top/bottom if everything can/should be string or if certain fields need to be in a certain format.
